sdk/storage/Azure.Storage.Common/src/Shared/AesGcm/AesGcmWindows.cs

```diff
@@ -0,0 +1,146 @@
+
+        internal AesGcmWindows(ReadOnlySpan<byte> key)
+        {
+            ThrowIfNotSupported();
+
+            Errors.CheckCryptKeySize(key.Length);
+            ImportKey(key);
+        }
+
+        internal AesGcmWindows(byte[] key)
+        {
+            ThrowIfNotSupported();
+
+            Errors.ThrowIfParamNull(key, nameof(key));
+
+            Errors.CheckCryptKeySize(key.Length);
+            ImportKey(key);
+        }
+
+        internal void Encrypt(byte[] nonce, byte[] plaintext, byte[] ciphertext, byte[] tag, byte[] associatedData = null)
+        {
+            Errors.ThrowIfParamNull(nonce, nameof(nonce));
+            Errors.ThrowIfParamNull(plaintext, nameof(plaintext));
+            Errors.ThrowIfParamNull(ciphertext, nameof(ciphertext));
+            Errors.ThrowIfParamNull(tag, nameof(tag));
+
+            Encrypt((ReadOnlySpan<byte>)nonce, plaintext, ciphertext, tag, associatedData);
+        }
+
+        internal void Encrypt(
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> plaintext,
+            Span<byte> ciphertext,
+            Span<byte> tag,
+            ReadOnlySpan<byte> associatedData = default)
+        {
+            CheckParameters(plaintext, ciphertext, nonce, tag);
+            EncryptCore(nonce, plaintext, ciphertext, tag, associatedData);
+        }
+
+        internal void Decrypt(byte[] nonce, byte[] ciphertext, byte[] tag, byte[] plaintext, byte[] associatedData = null)
+        {
+            Errors.ThrowIfParamNull(nonce, nameof(nonce));
+            Errors.ThrowIfParamNull(ciphertext, nameof(ciphertext));
+            Errors.ThrowIfParamNull(tag, nameof(tag));
+            Errors.ThrowIfParamNull(plaintext, nameof(plaintext));
+
+            Decrypt((ReadOnlySpan<byte>)nonce, ciphertext, tag, plaintext, associatedData);
+        }
+
+        internal void Decrypt(
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> ciphertext,
+            ReadOnlySpan<byte> tag,
+            Span<byte> plaintext,
+            ReadOnlySpan<byte> associatedData = default)
+        {
+            CheckParameters(plaintext, ciphertext, nonce, tag);
+            DecryptCore(nonce, ciphertext, tag, plaintext, associatedData);
+        }
+
+        private static void CheckParameters(
+            ReadOnlySpan<byte> plaintext,
+            ReadOnlySpan<byte> ciphertext,
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> tag)
+        {
+            if (plaintext.Length != ciphertext.Length)
+                throw new ArgumentException("SR.Cryptography_PlaintextCiphertextLengthMismatch");
+
+            if (!nonce.Length.IsLegalSize(NonceByteSizes))
+                throw new ArgumentException("SR.Cryptography_InvalidNonceLength", nameof(nonce));
+
+            if (!tag.Length.IsLegalSize(TagByteSizes))
+                throw new ArgumentException("SR.Cryptography_InvalidTagLength", nameof(tag));
```

Comment:
For many of these exception messages you should be able to get away with asserts, as it's non-public API in your copy.  But things like a tag mismatch (data tampering, or a wrong key or wrong nonce) you might want to actually populate.

All of the `SR.something` values are defined in the same file. 
 This one is at https://github.com/dotnet/runtime/blob/032a7dcbe1056493e8bab51e6b5b9503de727273/src/libraries/System.Security.Cryptography/src/Resources/Strings.resx#L417-L419, but they're all in there... generally alphabetically.